### DISCUSSION TITLE
Adding no slip condition on an hydrofoil

### DISCUSSION THREAD

**Discussion by julesangebault** [lululxvi/deepxde#880] [Q&A] [answered]
Hi, 
I'm trying to simulate the free surface flow caused by an hydrofoil moving near the free surface. To do so, I need to set the no slip Boundary condition on the surface of the hydrofoil. But for the hydrofoil geometry (naca0012), I'm using the `dde.geometry.PointCloud` and then the `dde.icbc.boundary_conditions.DirichletBC` But the dde.geometry.PointSet doesn't support on_boundary. Do you have an idea how I should proceed ?

**Comment by praksharma** [comment, marked as answer]
Use `dde.icbc.NeumannBC` . Or if you check the source code [here](https://github.com/lululxvi/deepxde/blob/2ee292f0cbfb0aed69a786b02bc062a938907207/deepxde/icbc/boundary_conditions.py#L81). You can use [`OperatorBC()`](https://github.com/lululxvi/deepxde/blob/2ee292f0cbfb0aed69a786b02bc062a938907207/deepxde/icbc/boundary_conditions.py#L135) for temporal problems. Check [this](https://github.com/lululxvi/deepxde/blob/7322b23d1f74cfda62fcfb12bfa24d879e12f578/examples/pinn_forward/wave_1d.py#L41) as an example.
